src/main/scala/com/evolution/scache/CacheMetered.scala

```diff
@@ -64,18 +64,23 @@ object CacheMetered {
           for {
             result <- cache.getOrUpdate1(key) {
               for {
-                start    <- MeasureDuration[F].start
-                value    <- value.attempt
-                duration <- start
-                _        <- metrics.load(duration, value.isRight)
-                value    <- value.liftTo[F]
+                _          <- metrics.get(false)
+                start      <- MeasureDuration[F].start
+                value      <- value.attempt
+                duration   <- start
+                loadSucceed = value match {
+                  case Right(_) | Left(Cache.NoneError) => true
+                  case Left(_)                          => false
+                }
+                _     <- metrics.load(duration, loadSucceed)
+                value <- value.liftTo[F]
               } yield {
                 val (a, v, release) = value
                 val release1 = releaseMetered(start, release.getOrElse { ().pure[F] })
                 (a, v, release1.some) // TODO is this a good idea to convert option to always some?
               }
             }
-            _     <- metrics.get(result.isRight)
+            _ <- metrics.get(true).whenA(result.isRight)
```

Comment:
could you please explain this?

Reply:
`getOrUpdate1` either use provided `F[_]` or not.
If we get `Left` - we missed the cache and the provided computation actually gets executed.
If we get `Right` - we hit the cache and the provided computation will never be executed.

The tricky moment here is that the provided computation might actually raise an exception
```scala
value <- value.liftTo[F] // might raise Cache.NoneError
```
or get cancelled. Given that we missed the cache tried to perform some work because of that but never updated metrics.

So my idea here is to call `metrics.get(hit = false)` in the provided computation otherwise to call `metrics.get(hit = true)` afterwards.

Reply:
edgy case, but a valid point.